tests/Doctrine/Tests/DBAL/Functional/PrimaryReadReplicaConnectionTest.php

```diff
@@ -0,0 +1,243 @@
+<?php
+
+namespace Doctrine\Tests\DBAL\Functional;
+
+use Doctrine\DBAL\Connections\PrimaryReadReplicaConnection;
+use Doctrine\DBAL\Driver\Statement;
+use Doctrine\DBAL\DriverManager;
+use Doctrine\DBAL\Schema\Table;
+use Doctrine\Tests\DbalFunctionalTestCase;
+use Throwable;
+
+use function array_change_key_case;
+use function sprintf;
+use function strlen;
+use function strtolower;
+use function substr;
+
+use const CASE_LOWER;
+
+/**
+ * @group DBAL-20
+ */
+class PrimaryReadReplicaConnectionTest extends DbalFunctionalTestCase
+{
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $platformName = $this->connection->getDatabasePlatform()->getName();
+
+        // This is a MySQL specific test, skip other vendors.
+        if ($platformName !== 'mysql') {
```

Comment:
@stof said this was supposed to work with PG too, should we maybe change this line?

Reply:
@greg0ire yes thats a future todo imho, the test actually only works with MySQL because of the charset test.